healthz/middleware.py

```diff
@@ -1,18 +1,41 @@
 from django.utils.deprecation import MiddlewareMixin
-from django.http import HttpResponse, HttpResponseServerError
+from django.http import HttpResponse, HttpResponseServerError, HttpResponseNotAllowed
+from django.conf import settings
 from logging import getLogger
 
 
 logger = getLogger('django')
 
 
 class HealthCheckMiddleware(MiddlewareMixin):
+    readiness_checks = None
+
     def process_request(self, request):
-        if request.method == "GET":
-            if request.path == "/readiness":
-                return self.readiness(request)
-            elif request.path == "/healthz":
-                return self.healthz(request)
+        if request.method != "GET":
+            return HttpResponseNotAllowed()
+
+        if not self.is_settings_defined():
```

Comment:
I would think default to "OK" rather than down when no checks are defined would be better? Not sure though 🤔 

I don't think it is necessary to check if settings were defined for the readiness checks. It could just behave like the healthz path if no readiness checks are defined. Just feels lime a more sane default.  No checks defined... no issues 🤔

Reply:
No probs. I was torn between both ways anyway. Will update it 👍

Reply:
The alternative I thought of was to use, 'database', 'cache', 'queues' as a key and provide a method, so similar to what we do with authentication classes for instance.